File: My_Notes/Programming_Note/Samples/async_samples/C#/Async/Async/Async1.cs
using System.Threading.Tasks;

namespace Async
{
    partial class Program
    {
        static async void Async1()
        {
            var job = new Job(3, 1, 2);
            var dbEngineer = new DBEngineer();
            var apiEngineer = new APIEngineer();
            var uiEngineer = new UIEngineer();
            Task task1 = dbEngineer.DoAsync(job);
            Task task2 = apiEngineer.DoAsync(job);
            Task task3 = uiEngineer.DoAsync(job);
            await Task.WhenAll(task1, task2, task3);
        }
    }
}

File: My_Notes/Programming/Samples/async_samples/dart/async1.dart
import 'dart:io';
import './job.dart';
import './engineer.dart';

void async1(){
  Job job = Job(3, 1, 2);
  DBEngineer dbEngineer = DBEngineer();
  APIEngineer apiEngineer = APIEngineer();
  UIEngineer uiEngineer = UIEngineer();
  Future<void> task1 = dbEngineer.doTaskAsync(job);
  Future<void> task2 = apiEngineer.doTaskAsync(job);
  Future<void> task3 = uiEngineer.doTaskAsync(job);

  Future.wait({task1, task2, task3});
}